File: adapter/php/adapter/MediaAdapter.php
<?php
namespace Adapter;

require_once 'interfaces/MediaPlayer.php';
require_once 'legacy/AdvancedMediaPlayer.php';
require_once 'concrete/VlcPlayer.php';
require_once 'concrete/Mp4Player.php';

use Interfaces\MediaPlayer;
use Legacy\AdvancedMediaPlayer;
use Concrete\VlcPlayer;
use Concrete\Mp4Player;

// Clase Adapter
class MediaAdapter implements MediaPlayer {
    private $advancedMediaPlayer;

    public function __construct($audioType) {
        if ($audioType == "vlc") {
            $this->advancedMediaPlayer = new VlcPlayer();
        } else if ($audioType == "mp4") {
            $this->advancedMediaPlayer = new Mp4Player();
        }
    }

    public function play($audioType, $fileName)
    {
        if ($audioType == "vlc") {
            $this->advancedMediaPlayer->playVlc($fileName);
        } else if ($audioType == "mp4") {
            $this->advancedMediaPlayer->playMp4($fileName);
        }
    }
}

File: adapter/python/media_adapter.py
from media_player import MediaPlayer
from advanced_media_player import VlcPlayer, Mp4Player

# Clase Adapter
class MediaAdapter(MediaPlayer):
    def __init__(self, audio_type):
        self.advanced_player = None
        
        if audio_type.lower() == "vlc":
            self.advanced_player = VlcPlayer()
        elif audio_type.lower() == "mp4":
            self.advanced_player = Mp4Player()
            
    def play(self, audio_type, file_name):
        if audio_type.lower() == "vlc":
            self.advanced_player.play_vlc(file_name)
        elif audio_type.lower() == "mp4":
            self.advanced_player.play_mp4(file_name)